### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by amritghimire**

### CODE REVIEW

File: .github/workflows/cargo.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    branches:
+      - main  # Set a branch name to trigger deployment
+  pull_request:
+
+jobs:
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
Comment (skshetry) [inline .github/workflows/cargo.yml]:
```suggestion
      - uses: actions/checkout@v3
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+.PHONY: build
```
Comment (skshetry) [inline Makefile]: Do we need a makefile?

`make check` is equivalent to `cargo check`, etc. `all` maybe, but is it worth maintaining this?

File: .github/workflows/cargo.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
+name: Various cargo commands
+
+on:
+  push:
+    branches:
+      - main  # Set a branch name to trigger deployment
+  pull_request:
+
+jobs:
+  check:
```
Comment (skshetry) [inline .github/workflows/cargo.yml]: Unless I am missing something, looks like we can use `matrix` here. something like following?
```yaml
strategy:
    matrix:
        commands: [fmt, build, test, check]
```

Comment (amritghimire) [reply .github/workflows/cargo.yml]: I haven't tried it yet. Can you add it?

File: Makefile (hunk 2 above)
Comment (amritghimire) [reply Makefile]: It helps to run all cargo commands at once.

**Comment by skshetry** [comment]
Thanks @amritghimire.

**Pull request merged**